dashboard/src/components/Config/AppRepoList/AppRepoForm.test.tsx

```diff
@@ -60,6 +67,54 @@ it("should not call the install method when the validation fails unless forced",
   expect(install).toHaveBeenCalled();
 });
 
+it("should not show the docker registry credentials section if the namespace is the global one", () => {
+  const wrapper = shallow(
+    <AppRepoForm {...defaultProps} kubeappsNamespace={defaultProps.namespace} />,
+  );
+  expect(wrapper.text()).not.toContain("Associate Docker Registry Credentials");
+});
+
+it("should render the docker registry credentials section", () => {
+  const wrapper = shallow(<AppRepoForm {...defaultProps} />);
+  expect(wrapper.find(AppRepoAddDockerCreds)).toExist();
+});
+
+it("should call the install method with the selected docker credentials", async () => {
+  const validate = jest.fn(() => true);
+  const install = jest.fn(() => true);
+  const wrapper = shallow(<AppRepoForm {...defaultProps} validate={validate} onSubmit={install} />);
+  wrapper.setState({ selectedImagePullSecrets: { "repo-1": true } });
+
+  const button = wrapper.find("form");
+  button.simulate("submit", { preventDefault: jest.fn() });
+  // wait for async functions
+  await wait(1);
```

Comment:
ugh - that's going to suck to maintain. Is this because it doesn't help to `await button.simulate` etc?

Reply:
well, it's not that difficult to maintain (the previous approach I had was worse, see for example https://github.com/kubeapps/kubeapps/pull/1684/files#diff-19c2034062c1971cbf14e6f364bcef47L184). It's just one line that allows to wait for mocked async functions to finish before evaluating tests results. Note that this is only needed when the Component code calls to something `async` and we wait for the result to update the state.

> Is this because it doesn't help to await button.simulate etc?

Yes, `button.simulate` doesn't return a promise so waiting that does nothing.

Reply:
This seems to handle the issue via `setImmediate`, which apparently though supported in node (so should work fine here) isn't going to be a standard? https://github.com/enzymejs/enzyme/issues/823#issuecomment-492984956

Reply:
yep, that doesn't seem that good either, at least now it's clear what it's doing.